### PR TITLE
Update "How to build" and MSSQL instructions

### PR DESCRIPTION
**Pull request by fenek**
This PR adds extra details about building MIM from source, testing it and connecting to MSSQL.

### CODE REVIEW

**Comment by NelsonVides** [comment]
I can confirm that the instructions worked on an Ubuntu Xenial VM but not on my main Ubuntu Disco laptop.

However, in Disco, mim detects the connection, if I go into the debug console, `eodbc:connect` does work, and the queries do return stuff. Where I manage to narrow the problem was on login a user, it fails for all of them: inside `ejabberd_auth_rdbms:check_password_wo_escape/4`, `rdbms_queries:get_password/2` always returns the second branch, the non-matching one.

**Comment by fenek** [comment]
@NelsonVides 

That's why I wrote for now that only Xenial works with MSSQL. It's most probably a bug in eodbc/odbc (I suspect that something changed in FreeTDS protocol, despite using the same version) and we need to update/debug it.

File: doc/operation-and-maintenance/known-issues.md [hunk 1]
```diff
@@ -10,3 +10,16 @@ MongooseIM will not connect to MySQL over TLS on OTP 20.3 due to [the MySQL driv
 * Upgrade OTP to 21.0 or higher.
 * Use unencrypted communication with MySQL.
 
+## MSSQL connectivity via ODBC
+
+ODBC driver currently used by MongooseIM is known to work only with Ubuntu Xenial x64 and FreeTDS 0.91-6.1build1.
+
+It definitely doesn't work on Ubuntu Bionic, CentOS 7 and macOS Mojave with the latest FreeTDS version.
```
Comment (NelsonVides) [inline doc/operation-and-maintenance/known-issues.md]: I'm not that sure about the harshness of `definitely` there. I'd rather go for `We've tested negatively on...` or simply omitting that `definitely` from there.

File: doc/operation-and-maintenance/known-issues.md [hunk 2]
```diff
@@ -10,3 +10,16 @@ MongooseIM will not connect to MySQL over TLS on OTP 20.3 due to [the MySQL driv
 * Upgrade OTP to 21.0 or higher.
 * Use unencrypted communication with MySQL.
 
+## MSSQL connectivity via ODBC
+
+ODBC driver currently used by MongooseIM is known to work only with Ubuntu Xenial x64 and FreeTDS 0.91-6.1build1.
```
Comment (NelsonVides) [inline doc/operation-and-maintenance/known-issues.md]: Perhaps this is a case of an article drop -> `The ODBC driver...`.

File: doc/developers-guide/Testing-MongooseIM.md [hunk 3]
```diff
@@ -1,9 +1,45 @@
...
+$ sudo apt install freetds-dev tdsodbc
+
+# CentOS
+$ sudo yum install freetds
+
+# macOS
+$ brew install freetds
+```
+
+Then, please modify `tools/travis-setup-db.sh` script to use proper FreeTDS paths.
```
Comment (NelsonVides) [inline doc/developers-guide/Testing-MongooseIM.md]: `to use the proper FreeTDS...`

File: doc/user-guide/How-to-build.md [hunk 4]
```diff
@@ -1,71 +1,116 @@
 # How to build MongooseIM
 
+Instructions provided in this page are verified for:
+
+* CentOS 7
+* Ubuntu 16.04 LTS (Xenial)
+* Ubuntu 18.04 LTS (Bionic)
+* macOS 10.14 (Mojave)
+
+For remaining OS versions the instructions should work as well, however some steps or file paths may be different.
```
Comment (NelsonVides) [inline doc/user-guide/How-to-build.md]: I'd go for `For other OS versions...`. "Remaining" makes it sound like the list of not-stated OS versions is short and has been clarified before (like, we've given a list of OSs before and we're now enumerating just a selection of them).

File: doc/developers-guide/Testing-MongooseIM.md [hunk 5]
```diff
@@ -1,9 +1,45 @@
 # Test runner
 
-**Warning: Some examples in this guide use the MySQL database. Please keep in mind that MongooseIM cannot connect to [MySQL via TLS on OTP 20.3](../operation-and-maintenance/known-issues.md).**
-
 The test runner script is used to compile MongooseIM and run tests.
 
+## IMPORTANT!
+
+* Some examples in this guide use the MySQL database. Please keep in mind that MongooseIM cannot connect to [MySQL via TLS on OTP 20.3](../operation-and-maintenance/known-issues.md).
+* ODBC preset may be tested [only on Ubuntu Xenial x64](../operation-and-maintenance/known-issues.md).
```
Comment (AnthonyJeannot) [inline doc/developers-guide/Testing-MongooseIM.md]: ODBC preset can only be tested on Ubuntu Xenial x64.

File: doc/operation-and-maintenance/known-issues.md (hunk 2 above)
Comment (AnthonyJeannot) [reply doc/operation-and-maintenance/known-issues.md]: The ODBC driver currently used by MongooseIM is known to only work with Ubuntu Xenial x64 and FreeTDS 0.91-6.1build1.

File: doc/operation-and-maintenance/known-issues.md (hunk 1 above)
Comment (AnthonyJeannot) [reply doc/operation-and-maintenance/known-issues.md]: It does not currently work on Ubuntu Bionic, CentOS 7 and macOS Mojave with the latest FreeTDS version.

File: doc/user-guide/How-to-build.md (hunk 4 above)
Comment (AnthonyJeannot) [inline doc/user-guide/How-to-build.md]: For any other OS versions, the instructions should still work, however, some steps or file paths may be different.

File: doc/user-guide/How-to-build.md [hunk 6]
```diff
@@ -1,71 +1,116 @@
...
+You may use a different package manager but you'll need to figure out the package names and file paths on your own.
 
-* C and C++ compiler
-* Git
-* Erlang/OTP 18.3 or higher
-* OpenSSL
-* ODBC
+### Step 2
 
-We recommend using [Homebrew](http://brew.sh) to manage packages on your Mac.
```
Comment (AnthonyJeannot) [inline doc/user-guide/How-to-build.md]: We recommend using Homebrew to manage the packages on your Mac.

File: doc/user-guide/How-to-build.md [hunk 7]
```diff
@@ -1,71 +1,116 @@
...
-Additionally you'll need to export some environment variables to point the compiler to OpenSSL
-installed with Homebrew:
+Please remember to replace `[installed version]` with the one that is present in your file system.
+1.0.2s was the most recent one when this guide was written.
+
+Now, please proceed to the "Building" section.
+
+## Preparing CentOS environment
+
+Please install required dependencies:
```
Comment (AnthonyJeannot) [inline doc/user-guide/How-to-build.md]: Please install the required dependencies:

File: doc/user-guide/How-to-build.md [hunk 8]
```diff
@@ -1,71 +1,116 @@
...
+$ sudo yum install git make zlib-devel unixODBC-devel gcc gcc-c++ erlang
 ```
 
-After you've done this, follow the steps from the [Compiling on *nix systems](#compiling-on-42nix-systems) section.
+Now, please proceed to the "Building" section.
+
+## Preparing Ubuntu environment
 
-## Compiling on \*nix systems
+Please install required dependencies:
```
Comment (AnthonyJeannot) [inline doc/user-guide/How-to-build.md]: Please install the required dependencies:

File: doc/user-guide/How-to-build.md [hunk 9]
```diff
@@ -1,71 +1,116 @@
...
+```bash
+$ sudo apt install git make zlib1g-dev unixodbc-dev gcc g++ erlang
+```
 
+Now, please proceed to the "Building" section.
 
-        $ ./rebar3 compile
+## Building
 
-To generate a full MongooseIM release (i.e. an executable with mysql, pgsql or other dependencies):
```
Comment (AnthonyJeannot) [inline doc/user-guide/How-to-build.md]: To generate a full MongooseIM release (i.e. one that is executable with mysql, pgsql or other dependencies):

File: doc/user-guide/How-to-build.md [hunk 10]
```diff
@@ -82,32 +127,41 @@ To run MongooseIM from the project tree after compiling it, change to `$REPO/_bu
...
+    uptime: 0 days 00:00:12
+    distribution protocol: inet_tcp
+    logs:
         log/ejabberd.log
+```
+
 
 ## Building the testing target and running tests
 
 For testing purposes there's a different make target available:
```
Comment (AnthonyJeannot) [inline doc/user-guide/How-to-build.md]: Hey, I don't understand 'make target' is that a developer specific term everyone will understand?

Comment (NelsonVides) [reply doc/user-guide/How-to-build.md]: @AnthonyJeannot yeap, `make` is a developer tool, every developer knows it.

**Pull request merged**